crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -303,51 +299,66 @@ use std::{
 /// # bevy_ecs::system::assert_is_system(my_system);
 /// ```
 pub trait WorldQuery {
-    type Fetch: for<'world, 'state> Fetch<'world, 'state, State = Self::State>;
-    type State: FetchState;
-    type ReadOnlyFetch: for<'world, 'state> Fetch<'world, 'state, State = Self::State>
-        + ReadOnlyFetch;
+    type State: FetchState + for<'w> FetchInit<'w>;
+
+    fn shrink<'wlong: 'wshort, 'wshort>(item: QueryItem<'wlong, Self>) -> QueryItem<'wshort, Self>;
 }
 
-pub type QueryItem<'w, 's, Q> = <<Q as WorldQuery>::Fetch as Fetch<'w, 's>>::Item;
+pub type QueryFetch<'w, Q> = <<Q as WorldQuery>::State as FetchInit<'w>>::Fetch;
+pub type QueryItem<'w, Q> = <<Q as WorldQuery>::State as FetchInit<'w>>::Item;
+pub type ROQueryFetch<'w, Q> = <<Q as WorldQuery>::State as FetchInit<'w>>::ReadOnlyFetch;
+pub type ROQueryItem<'w, Q> = <<Q as WorldQuery>::State as FetchInit<'w>>::ReadOnlyItem;
+
+pub trait FetchInit<'world>: FetchState {
```

Comment:
`FetchInit` contains no methods and `FetchState` has an `init` method probably will add to the difficulty distinguishing between these two types.

I get that the two types are useful and am not questioning the split but I don't look at `FetchInit` and `FetchState` and think "we want the `Fetch` provided by `FetchInit` to avoid being tied to a particular lifetime" and the existence of `FetchState::init` certainly muddies the waters.

To avoid critiquing names without providing any alternatives I will say `FetchDetails` as an alternative but I don't know it is any better lol.